app/supportAssignment/services/supportActions.js

```diff
@@ -0,0 +1,514 @@
+			});
+		},
+		updateSupportAppointment: function (supportAppointment) {
+			supportAppointment.percentage = parseFloat(supportAppointment.percentage);
+			supportAppointment.scheduleId = supportReducer._state.schedule.id;
+			supportAppointment.termCode = supportReducer._state.ui.termCode;
+
+			supportService.updateSupportAppointment(supportAppointment).then(function(payload) {
+				$rootScope.$emit('toast', { message: "Updated Appointment", type: "SUCCESS" });
+				supportReducer.reduce({
+					type: UPDATE_SUPPORT_APPOINTMENT,
+					payload: {
+						supportAppointment: payload
+					}
+				});
+			}, function (err) {
+				$rootScope.$emit('toast', { message: "Could not update teaching assistants.", type: "ERROR" });
+			});
+		},
+		// Example 'Comments', 'Teaching Assignments'
+		setViewPivot: function (tabName) {
+			supportReducer.reduce({
+				type: SET_VIEW_PIVOT,
+				payload: {
+					tabName: tabName
+				}
+			});
+		},
+		// Example 'Reader', 'Teaching Assistants'
+		setViewType: function (viewType) {
+			supportReducer.reduce({
+				type: SET_VIEW_TYPE,
+				payload: {
+					viewType: viewType
+				}
+			});
+		},
+		setSupportStaffTab: function(tabName, supportStaffId) {
+			supportReducer.reduce({
+				type: SET_SUPPORT_STAFF_TAB,
+				payload: {
+					tabName: tabName,
+					supportStaffId: supportStaffId
+				}
+			});
+		},
+		updateTableFilter: function (query) {
+			var action = {
+				type: UPDATE_TABLE_FILTER,
+				payload: {
+					query: query
+				}
+			};
+			supportReducer.reduce(action);
+		},
+		openAvailabilityModal: function(supportStaff) {
+			supportReducer.reduce({
+				type: OPEN_AVAILABILITY_MODAL,
+				payload: {
+					supportStaff: supportStaff
+				}
+			});
+		},
+		closeAvailabilityModal: function(supportStaff) {
+			supportReducer.reduce({
+				type: CLOSE_AVAILABILITY_MODAL,
+				payload: {}
+			});
+		},
+		performInitCalculations: function() {
+			this.calculateSectionGroupScheduling();
+			this.calculateSectionScheduling();
+			this.calculateScheduleConflicts();
+			this.calculateStaffAssignmentOptions();
+		},
+
+		calculateScheduleConflicts: function() {
+			var self = this;
+
+			var conflicts = {
+				bySupportStaffId: {},
+				bySectionId: [],
+				bySectionGroupId: [],
+			};
+
+			supportReducer._state.supportStaffList.ids.forEach(function(supportStaffId) {
+				var supportStaff = supportReducer._state.supportStaffList.list[supportStaffId];
+				var supportStaffResponse = supportReducer._state.supportStaffSupportCallResponses.bySupportStaffId[supportStaffId];
+
+				var supportConflicts = conflicts.bySupportStaffId[supportStaffId] = {
+					sectionIds: [],
+					sectionGroupIds: []
+				};
+
+				// If the support staff has no availability set, then there cannot be conflicts
+				if (!supportStaffResponse || !supportStaffResponse.availabilityBlob || supportStaffResponse.availabilityBlob.length == 0) {
```

Comment:
Write the ! as == false or == null or == undefined (whichever is correct) for readability.